model/cloudfoundry/resources.yaml

```
@@ -0,0 +1,51 @@
+groups:
+  - id: cloudfoundry
+    type: resource
+    name: cloudfoundry
+    brief: >
+      A CloudFoundry application or system component.
+    attributes:
+      - ref: cloudfoundry.source.type
+        requirement_level:
+          conditionally_required: if the the event is a log message
```

from pure otel side, `event` and `log` are meaningful terms (events are a subset of logs), so this looks confusing. Maybe this would work (if it's correct):

```suggestion
          conditionally_required: if the the source produces log messages
```

Since resource attributes are applied globally to all telemetry reported by a trace/metric/log provider, individual telemetry items (logs) can't have different resource attributes. If source type is a property of a specific log, you may want to define it in a separate `attribute_group` or describe the whole event semantic convention including this attribute.

---

This attribute should only be added to logs, not to metrics or spans. It is constant for the service for all log messages. I understand, that this is not the correct way to model this as a resource attribute. I will remove this attribute from this PR.

@lmolkova: What would be your proposal to include this attribute in log messages? Should this be a separate convention just for this log attribute?

---

If it's a constant for all logs emitted by the service it can be added as a resource attribute and then populated only on the [logger provider](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/sdk.md#loggerprovider) but not spans or metrics - that's totally fine.
It'd be great to explain it in the note/brief so people know what is it and how to use it.

If you're interested in describing it as a log attribute, please take a look at some log examples, e.g. [exceptions](https://github.com/open-telemetry/semantic-conventions/blob/b780f480e73ffc2d1da117689475b7b9c8af610b/docs/exceptions/exceptions-logs.md#attributes).

---

I suggest to leave this attribute out for now. It will most likely be populated only inside the CloudFoundry system components by an OTel Collector. As indicated, this field is hard to fill for application developers trying to instrument their own application. Therefore, I think it is better to leave it out and add it at a later point in time, when it is clearer, how it is added and that it is still required in a world, where CloudFoundry primarily uses OpenTelemetry for its observability signals.